botfront/imports/lib/utils.js

```diff
@@ -28,54 +23,26 @@ Meteor.callWithPromise = (method, ...myParameters) => new Promise((resolve, reje
 });
 
 export const formatError = (error) => {
-    const { response } = error;
+    if (error instanceof Meteor.Error) return error;
+
+    const {
+        response, request, code, message,
+    } = error;
+    
     if (response && response.status) {
         // axios error
-        return formatAxiosError('HTTP', error);
+        return new Meteor.Error(error.response.status, error.response.data.error);
     }
-
-    if (error.code && error.code === 11000) {
-        return new Meteor.Error('Duplicate key', error.errmsg);
+    if (request && code === 'ECONNREFUSED') {
+        // axios error
+        return new Meteor.Error('500', `Could not reach host at ${error.config.url}`);
     }
 
-    const { name, error: type, message } = error;
-    if ((type || name) && message) {
-        if (type) {
-            // regular meteor error
-            return error;
-        } if (name) {
-            // mongo error
-            return new Meteor.Error('mongo', `${type}: ${message}`);
-        }
-    } else if (message) {
-        // some other kind of error
-        return new Meteor.Error('error', message);
+    if (code === 11000) {
+        return new Meteor.Error('Duplicate key', error.errmsg);
     }
 
-
-    // fallback error
-    console.log(error);
-    return new Meteor.Error('unknown', 'Something went wrong');
-};
-
-export const getAxiosError = (e) => {
-    console.log(e);
-    // TODO identify all possible error object structure and create an external function to derive code and message consistently
-    let error = null;
-    if (e.response) {
-        // The request was made and the server responded with a status code
-        // that falls out of the range of 2xx
-        error = new Meteor.Error(e.response.status, e.response.data.error);
-    } else if (e.request) {
-        // The request was made but no response was received
-        // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
-        // http.ClientRequest in node.js
-        error = new Meteor.Error('399', 'timeout');
-    } else {
-        // Something happened in setting up the request that triggered an Error
-        error = new Meteor.Error('500', e.message);
```

Comment:
I feel like those case are not treated anymore, in the new function, may be we could improve them and treat the most commoms case. Lik giving back a readable error saying that the credentials are bad or that the  instance adress is not correct ?

Reply:
Everything is there, just removed the '399 timeout' since that's spurious. Now it says `Could not reach host at ${error.config.url}` instead of '399 Timeout'